blastai/tools.py

```diff
@@ -196,7 +196,7 @@ async def get_first_subtask_result(
                     success=False,
                     error="No valid task IDs provided (filtered out current task)"
                 )
-            return await self._get_first_subtask_result(scheduler, task_ids, as_final=False)
+            return await self._get_first_subtask_result(scheduler, task_ids, as_final=True)
```

Comment:
I noticed subtasks being repeatedly launched (very rarely, but does happen sometimes) with this set to `False`